### PR TITLE
Add a nice handler for a change to location instead of just making an error

### PR DESCRIPTION
**Pull request by devsnek**

### CODE REVIEW

File: lib/jsdom/living/window/Location-impl.js [hunk 1]
```diff
@@ -11,15 +11,21 @@ const navigate = require("./navigation.js").navigate;
...
     this.url = null;
   }
 
   get _url() {
-    return this._relevantDocument._URL;
+    return this._basicURL || this._relevantDocument._URL;
   }
 
   _locationObjectSetterNavigate(url) {
+    if (this._basicURL) {
```
Comment (devsnek) [inline lib/jsdom/living/window/Location-impl.js]: quick way to stop writes on basic locations, since they should technically be immutable

**Comment by ForbesLindesay** [comment]
I'm working on adding navigation support directly into jsdom (see #1902).  The issue with just adding a hook for it is that jsdom already handles history for fragment changes and push state.  I don't think it will take long for us to add basic navigation support to jsdom, which I think will be a better solution in the long run.

**Comment by devsnek** [comment]
@ForbesLindesay perhaps some of the ideas here can get merged into your pr, since I think the end user should be able to track exactly when a navigation occurs.

**Comment by ForbesLindesay** [comment]
@devsnek I think there will be an event fired when navigation occurs, once I finish what I'm working on.

**Comment by devsnek** [comment]
closing due to merge of #1905

**Pull request closed without merging**